treeherder/webapp/api/auth.py

```diff
@@ -16,3 +26,30 @@ def hawk_lookup(id):
         'key': str(credentials.secret),
         'algorithm': 'sha256'
     }
+
+
+class TaskclusterAuthViewSet(viewsets.ViewSet):
+
+    @list_route(methods=['get'])
+    def login(self, request):
+        """
+        Verify credentials with Taskcluster
+
+        """
+        authorization = request.META.get("HTTP_OTH", None)
+        host = request.query_params.get("host", None)
+        port = request.query_params.get("port", None)
+
+        user = authenticate(authorization=authorization, host=host, port=int(port))
+        login(request, user)
+
+        return Response({
+            "message": "login success",
+            "user": {"email": user.email,
+                     "is_staff": user.is_staff}
```

Comment:
Ah, so there's a bit of authorization going on here based on the clientId :(

Could we base that on a scope instead?  You could just check whether the user has the `assume:project:treeherder:staff` scope in the response from `authenticateHawk`.  It'd be fine to reflect that into the User object on every login, if you're comfortable with a user's staff-ness persisting until their TH login expires, even if they lose the scope (for example through an LDAP group change or becoming ex-staff)